indico/modules/events/registration/controllers/management/invitations.py

```diff
@@ -62,10 +88,79 @@ def _process(self):
+    """Send a reminder to pending invitees."""
+
+    @use_kwargs({
+        'from_address': fields.String(required=True, validate=not_empty),
+        'body': fields.String(required=True, validate=not_empty),
+        'subject': fields.String(required=True, validate=not_empty),
+        'bcc_addresses': fields.List(LowercaseString(validate=validate.Email())),
+        'copy_for_sender': fields.Bool(load_default=False),
+    })
+    def _process(self, from_address, body, subject, bcc_addresses, copy_for_sender):
+        if from_address not in self.event.get_allowed_sender_emails():
+            abort(422, messages={'from_address': ['Invalid sender address']})
+        invitations = _query_pending_invitations(self.regform)
+        for invitation in invitations:
+            email_body = replace_placeholders('registration-reminder-email', body, event=self.event,
+                                              invitation=invitation)
+            email_subject = replace_placeholders('registration-reminder-email', subject,
+                                                 event=self.event, invitation=invitation)
+            tpl = get_template_module('emails/custom.html', subject=email_subject, body=email_body)
+            bcc = {session.user.email} if copy_for_sender else set()
+            bcc.update(bcc_addresses)
+            with self.event.force_event_locale():
+                email = make_email(to_list=invitation.email, bcc_list=bcc, from_address=from_address,
+                                   template=tpl, html=True)
+            send_email(email, self.event, 'Registration')
+        return jsonify(count=len(invitations))
+
+
+class RHRegistrationFormRemindersMetadata(RHManageRegFormBase):
+    """Get the metadata for registration reminder emails."""
+
+    def _process(self):
+        with self.event.force_event_locale():
+            tpl = get_template_module('events/registration/emails/registration_reminder_default.html', event=self.event)
+            body = tpl.get_html_body()
+            subject = tpl.get_subject()
+        placeholders = get_sorted_placeholders('registration-reminder-email')
+        invitations = _query_pending_invitations(self.regform)
+        recipients = sorted([invitation.email for invitation in invitations])
+        return jsonify({
+            'senders': list(self.event.get_allowed_sender_emails().items()),
+            'recipients': recipients,
+            'body': body,
+            'subject': subject,
+            'placeholders': [p.serialize() for p in placeholders],
+        })
+
+
+class RHRegistrationFormRemindersPreview(RHManageRegFormBase):
+    """Preview a registration reminder email."""
+
+    @use_kwargs({
+        'body': fields.String(required=True),
+        'subject': fields.String(required=True),
+    })
+    def _process(self, body, subject):
+        invitations = _query_pending_invitations(self.regform)
+        if not invitations:
+            abort(404)
```

Comment:
Not a big fan of `abort()` for this - it shows 404 with a default "Not found" error which wouldn't be very clear. I think `BadRequest('No invitations')` would be clearer